src/main/java/mate.academy/Main.java

```diff
@@ -3,15 +3,34 @@
 import java.util.List;
 import mate.academy.lib.Injector;
 import mate.academy.model.Product;
+import mate.academy.service.FileReaderService;
+import mate.academy.service.ProductParser;
 import mate.academy.service.ProductService;
+import mate.academy.service.impl.FileReaderServiceImpl;
+import mate.academy.service.impl.ProductParserImpl;
+import mate.academy.service.impl.ProductServiceImpl;
 
 public class Main {
 
     public static void main(String[] args) {
-        // Please test your Injector here. Feel free to push this class as a part of your solution
         Injector injector = Injector.getInjector();
-        ProductService productService = null;
+
+        ProductService productService = (ProductServiceImpl)
+                injector.getInstance(ProductServiceImpl.class);
```

Comment:
Casting to the implementation 'ProductServiceImpl' is not a good practice in terms of dependency injection. You should program to an interface, not an implementation. Cast to 'ProductService' instead.